vector-im/element-android
# Fix lockscreen's 'device locked' crash on Android 12 and 12L devices

## Type of change


## Content

Fixes #6768.
Depends on #6769.

## Motivation and context

On Android 12 and 12L (APIs 31 and 32) there seems to be a bug that causes `KeyguardManager.isDeviceLocked` to be true for a few milliseconds even after the lock screen has been dismissed and the app is already displayed. 

Since we added `setRequiresDeviceUnlocked(true)` to the biometric key, it can only be used after the device is unlocked and `KeyStore` will throw an exception if we try to access the key while it's locked. In this case, it's causing crashes when trying to update `LockScreenViewModel`'s state based on the biometric keys info.

I also tried to simplify the implementation of the lock screen feature by removing some redundant components (LockScreenConfigurationProvider).

## Screenshots / GIFs





## Tests



- Make sure you're on Android 12 or 12L, in a physical device if possible (emulators on those APIs sometimes don't work well with biometric auth).
- Also, make sure the device has biometric authentication enabled.
- Open the app, set up biometric authentication for it if it wasn't already enabled. You can do this on Settings > Security and privacy > Protect access.
- With the app visible, lock and unlock the device using the power button then your pin/pattern/fingerprint.

If the app didn't crash, the fix is working.

## Tested devices

- OS version(s): 13, 12L, 12, 11.

## Checklist
